livekit_models.proto

```diff
@@ -167,3 +167,48 @@ message ClientInfo {
   string browser = 7;
   string browser_version = 8;
 }
+
+// server provided client configuration
+message RTCClientConfiguration {
+	AudioConfiguration audio = 1;
+  VideoConfiguration video = 2;
+  SignalConfiguration signal = 4;
+}
+
+message AudioConfiguration {
+  message AudioProcessing {
+    // echo cancellation
+    bool aec = 1;
+    // noise suppression
+    bool ns = 2;
+    // automatic gain control
+    bool agc = 3;
+  }
+  message AudioPreferences {
+    bool dtx = 1;
+    int32 sample_rate = 2;
+    int32 channel_count = 3;
+  }
+  Codec codec = 1;
+  AudioProcessing processing = 2;
+  AudioPreferences preferences = 3;
+
+  // TODO sample/playout config(micphone, speaker, earphone ...)
```

Comment:
Will defer to @cnderrauber for clarification.

I think this might be more to do with mobile devices. In general, I think this would something like a exclusion list. For example, it might be possible that certain devices are not supported. Even on desktop, I have run into WebRTC not able to support certain combination of OS + device. I am thinking we can use this object to specify such exclusion and potentially emit an info/warn event from SDK so that devs have the ability to display something like "Potentially unsupported device, your mileage may vary". Just a thought.

Reply:
On the video render side, I have seen `hardware_acceleration` flag in browsers. I think that is the difference of using GPU/WebGL (sorry I do not know the details how these things work, need to educate myself), discrete GPU vs integrated GPU etc. There is probably some combination of CPU + GPU + OS where disabling hardware acceleration will increase chances of things working. It is probably a large matrix. Just noting down stuff from experience :-)

Reply:
Yeah, it's about mobile devices that have more than one capture/playback devices that can't work  appropriately , we need to set device manually to make it work. Most of those cases is bluetooth earpods at android can't capture/playback at desired samplerate/channel count